wolfcrypt/src/wc_pkcs11.c

```diff
@@ -544,10 +530,36 @@ int wc_Pkcs11Token_Init(Pkcs11Token* token, Pkcs11Dev* dev, int slotId,
                 }
             }
             if (ret == 0) {
-                if (slotCnt > 0)
+                if (tokenName != NULL && tokenSz > 0) {
+                    /* match based on token name */
```

Comment:
Would like this in a separate function: Pkcs11Slot_FindByTokenName() or such.

Reply:
Done. See `Pkcs11Slot_FindByTokenName`.